src/site/mod.rs

```diff
@@ -45,48 +45,60 @@ impl Site {
             }
         }
 
-        if let Some(repo_url) = &config.repository {
-            let context = Context::new(repo_url, config.artifacts.cargo_dist)?;
-            if config.artifacts.has_some() {
-                let index = Page::index_with_artifacts(&context, &layout_template, config)?;
-                pages.push(index);
-                if context.latest_dist_release.is_some()
-                    || config.artifacts.package_managers.is_some()
-                {
-                    let body = artifacts::page(&context, config)?;
-                    let artifacts_page =
-                        Page::new_from_contents(body, "artifacts.html", &layout_template, config);
-                    pages.push(artifacts_page);
+        let mut index = Page::index(&layout_template, config)?;
```

Comment:
alternatively you can have index be an Option (= None here) and then at the end you `.unwrap_or_else` with this line, saving the cost of building the page when not needed..?